lib/modules/datasource/terraform-provider/index.ts

```diff
@@ -97,18 +113,37 @@ export class TerraformProviderDatasource extends TerraformDatasource {
     if (latestVersion) {
       latestVersion.releaseTimestamp = res.published_at;
     }
-    dep.homepage = `${registryURL}/providers/${repository}`;
-    logger.trace({ dep }, 'dep');
+    dep.homepage = `${registryUrl}/providers/${repository}`;
+    return dep;
+  }
+
+  /**
+   * this version uses the Provider Registry Protocol that all registries are required to implement
+   * https://www.terraform.io/internals/provider-registry-protocol
+   */
+  private async queryRegistryVersions(
+    serviceDiscovery: ServiceDiscoveryResult,
+    registryUrl: string,
+    repository: string
+  ): Promise<ReleaseResult> {
+    const backendURL = `${registryUrl}${serviceDiscovery['providers.v1']}${repository}/versions`;
+    const res = (await this.http.getJson<TerraformProviderVersions>(backendURL))
+      .body;
+    const dep: ReleaseResult = {
+      releases: res.versions.map(({ version }) => ({
+        version,
+      })),
+    };
     return dep;
   }
 
   // TODO: add long term cache (#9590)
   private async queryReleaseBackend(
     packageName: string,
-    registryURL: string
+    registryUrl: string
```

Comment:
please do those refactorings in separate PR's for easier review

Reply:
fine. removed it. I just figured so much has changed anyway in this file